Several changes and improvements

dufkan opened the pull request:
This pull request includes following changes:
- Fix of an issue causing `Keygen_ECC_0x0010.csv` to be empty on some TPMs
- `quicktest` command renamed to `capability`
- `fulltest` command renamed to `all`
- New command `extensive` runs all tests with extensive configuration
- `rng` output is now collected in binary file `Rng.bin`
- `all` command in the default setting collects 512 KiB from `rng`
- `extensive` command in the default setting collects 16 MiB from `rng`
- Measurement is accelerated by skipping configurations that fail three times (return three times non-zero return code)
- Unnecessary key exports in `cryptoops` are avoided
- Several outputs of the tool are unified (naming, deliminers in CSVs)
- Support new (non-deprecated) variant of TPM2_CC_EncryptDecrypt
- Basic system information collection
- Log message improvements
- Add `--only-measure` switch that skips formatting and post-processing of data
- Add `--include-legacy` switch that creates legacy result files

dufkan committed:
Add system information collection and update fulltest configuration

mbroz commented:
Hm, just a word of warning - DMI decode can contain serial number and Asset tag that can directly identify who owns the laptop. Just be sure we do not store more data that we really need...

Cweorth commented:
I checked the DMI decode source code:
https://github.com/mirror/dmidecode/blob/3882a45fb6b7a36ba9463ee53e46a9090ecb8d8c/dmidecode.c#L4326

As the data is collected as `dmidecode |grep -A3 '^System Information'`, it cannot contain Asset Tag nor Serial Number in *the current version*.

However, if it is crucial that we **never** obtain such information, the command could be changed so that only explicitly named attributes are retained from `dmidecode -t system`, as a (best effort) protection against future changes in dmidecode's output:

`dmidecode -t system | grep -Ei '^\s*(manufacturer|product name|version):'`

dufkan committed:
Change dmidecode command and remove stderr from output

dufkan commented:
Thanks to @mbroz for pointing out the potential issue and @Cweorth for providing the fix.

dufkan committed:
Change test naming, add extensive test configuration, refactor command handling

Command `quicktest` is changed to `capability`.
`Quicktest_*` files are renamed to `Capability_*`.
Command `fulltest` is changed to `all`.
New command with extensive test setting was added as `extensive`.

dufkan committed:
Fix get_tpm_id parsing issue

dufkan committed:
Fix keygen for BN_P256 with restricted usage scenarios

dufkan committed:
Skip remaining operations after reaching failure limit

dufkan committed:
Unify delimiters in output CSVs

dufkan committed:
Avoid repeated key extractions in cryptoops

dufkan committed:
Support new version of EncryptDecrypt command

dufkan committed:
Fix off by one in progress skipping

dufkan committed:
Improve log messages in perf

dufkan committed:
Concatenate rng outputs in Rng.bin

dufkan committed:
Fix performance statistics

dufkan committed:
Collect more rng data in 'all' configuration

dufkan committed:
Add --only-measure switch

dufkan committed:
Unify keygen output other commands

dufkan committed:
Rename results and performance files

dufkan committed:
Move status.c to algtest directory

dufkan committed:
Unify result and performace file formats

dufkan committed:
Bump version to 2.0

dufkan committed:
Add --include-legacy switch

dufkan committed:
Use local image tag in legacy header

dufkan committed:
Minor code improvements

dufkan committed:
Remove docker

mbroz commented on README.md:
````diff
@@ -4,30 +4,34 @@ It has been found out that implementation specifics of cryptographic smart cards
-The main keygen test generates 1000 RSA key pairs on the TPM for each available keylength (usually only two: 1024 and 2048). Depending on speed of generation, this should last up to few hours (2-3 hours with current TPMs) while consuming minimum system resources (the computation usually happens on a separate chip). If you cannot afford to let it run for too long, consider appending option `-d <du ...
+Build `tpm2-algtest` tool:
+```sh
+$ git clone https://github.com/dufkan/tpm2-algtest.git
````
Wrong repo link?

mbroz commented on algtest/util.c:
```diff
@@ -30,3 +30,17 @@ FILE *open_csv(const char *filename, const char *header)
+FILE *open_bin(const char *filename)
+{
+    char path[256];
+    strncpy(path, options.outdir, strlen(options.outdir) + 1);
```
This smells like possible overflow...  better PATH_MAX? (and perhaps snprintf for the whole size). Just saying :)

dufkan committed:
Fix repository url

dufkan committed:
Change file path concatenation
